lib/OpenQA/WebAPI/Plugin/Helpers.pm

```diff
@@ -220,15 +220,19 @@ sub register ($self, $app, $config) {
             return $c->tag('span', title => $text, $text);
         });
 
-    my @unfinished_states = (EXECUTION_STATES, PRE_EXECUTION_STATES);
+    my %progress_bar_query_by_key = (
+        unfinished => [state => [EXECUTION_STATES, PRE_EXECUTION_STATES]],
+        skipped => [result => [ABORTED_RESULTS]],
+        failed => [result => [FAILED, NOT_COMPLETE_RESULTS]],
+    );
     $app->helper(
         build_progress_bar_section => sub ($c, $key, $res, $max, $params, $class = '') {
             return '' unless $res;
             my $url = $params->{url};
             my $text = "$res $key";
             my $link_or_text = $url
               ? sub {
-                $url->query($key eq 'unfinished' ? (state => \@unfinished_states) : (result => $key));
+                $url->query(@{$progress_bar_query_by_key{$key} // [result => $key]});
```

Comment:
You'll need to update `t/ui/14-dashboard-parents.t`

Reply:
I was in fact hoping for a failing test to know where to extend/adapt them :-)